File: Scripts/micro_dao/avaliacao.py
from micro_dao.objeto_modelo import ObjetoModelo

class Avaliacao(ObjetoModelo):

    _id: int
    _nota: int
    _fk_servico_prestacao: int
    _fk_usuario: int

    def __init__(self, nota: int, fk_servico_prestacao: int, fk_usuario: int):
        self.set_nota(nota)
        self.set_fk_servico_prestacao(fk_servico_prestacao)
        self.set_fk_usuario(fk_usuario)

    def get_id(self) -> int:
        return self._id

    def set_id(self, id: int):
        self._id = id
        return self

    def get_nota(self) -> int:
        return self._nota

    def set_nota(self, nota: int):
        self._nota = nota
        return self

    def set_fk_servico_prestacao(self, fk_serv_prest: int):
        self._fk_servico_prestacao = fk_serv_prest
        return self

    def get_fk_servico_prestacao(self) -> float:
        return self._fk_servico_prestacao

    def get_fk_usuario(self) -> int:
        return self._fk_usuario

    def set_fk_usuario(self, fk_usuario: int):
        self._fk_usuario = fk_usuario
        return self

File: Projeto_Java/src/Dominio/Entidades/Avaliacao.java
package Dominio.Entidades;

import jdk.nashorn.internal.runtime.regexp.joni.exception.ValueException;

import javax.persistence.*;

@Entity
@Table(name = "avaliacao")
public class Avaliacao {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "nota")
    private int nota;

    @Column(name = "fk_usuario")
    private int fkUsuario;

    @Column(name = "fk_servico")
    private int fkServico;

    @OneToOne(fetch = FetchType.EAGER, targetEntity = Comentario.class)
    @JoinColumn(name = "id", referencedColumnName = "id")
    private Comentario comentario;

    public Avaliacao(){}

    public Avaliacao(int nota, int fkUsuario, int fkServico){
        this.nota = nota;
        this.fkUsuario = fkUsuario;
        this.fkServico = fkServico;
    }

    public Avaliacao(int nota, int fkUsuario, int fkServico, Comentario comentario){
        this.setNota(nota);
        this.setComentario(comentario);
        this.setFkUsuario(fkUsuario);
        this.setFkServico(fkServico);
    }

    public Avaliacao(int id, int nota, int fkUsuario, int fkServico,
                     Comentario comentario){
        this.setId(id);
        this.setNota(nota);
        this.setComentario(comentario);
        this.setFkUsuario(fkUsuario);
        this.setFkServico(fkServico);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getNota() {
        return nota;
    }

    public void setNota(int nota) {

        if (nota < 1) throw new ValueException("O campo '" + "nota" + "' deve conter valor maior ou igual a " + 1 + "'!");

        this.nota = nota;
    }

    public int getFkUsuario() {
        return fkUsuario;
    }

    public void setFkUsuario(int fkUsuario) {
        this.fkUsuario = fkUsuario;
    }

    public int getFkServico() {
        return fkServico;
    }

    public void setFkServico(int fkServico) {
        this.fkServico = fkServico;
    }

    public Comentario getComentario() {
        return comentario;
    }

    public void setComentario(Comentario comentario) {
        this.comentario = comentario;
    }

    @Override
    public String toString() {
        return String.format(
                "ID:\t\t\t%d;\nNota:\t\t%d;\nFK_usuario:\t%d;\nFK_servico:\t%d;" +
                        "\n\n---COMENTÁRIO---\n",
                this.getId(), this.getNota(), this.getFkUsuario(), getFkServico()) + (comentario != null ? comentario.toString() : "~NÃO HÁ COMENTÁRIO!");
    }
}
